nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/RunFlowMarkLogic.java

```diff
@@ -161,9 +164,12 @@ public void init(final ProcessorInitializationContext context) {
      */
     @OnScheduled
     public void onScheduled(ProcessContext context) {
-        DatabaseClientConfig clientConfig = context.getProperty(DATABASE_CLIENT_SERVICE)
-            .asControllerService(MarkLogicDatabaseClientService.class)
-            .getDatabaseClientConfig();
+        var controllerService = context.getProperty(DATABASE_CLIENT_SERVICE)
+            .asControllerService(MarkLogicDatabaseClientService.class);
+        if (controllerService == null) {
```

Comment:
TIL (after some searching) of `Objects.requireNonNull(thing, message)` - use that here instead for brevity. And throwing a NullPointerException here is appropriate - it really would be a connector error since NiFi should prevent this state from ever happening (since a controller service is required in order for the processor to be scheduled).